gitops/snapshot.go

```diff
@@ -152,17 +172,38 @@ func MarkSnapshotIntegrationStatusAsInProgress(adapterClient client.Client, ctx
 	if err != nil {
 		return nil, err
 	}
+
+	if helpers.HasLabel(snapshot, BuildPipelineRunFinishTimeLabel) {
+		snapshotInProgressTime := &metav1.Time{Time: time.Now()}
+		buildPipelineRunFinishTimeStr := snapshot.Labels[BuildPipelineRunFinishTimeLabel]
+		buildPipelineRunFinishTimeInt, _ := strconv.ParseInt(buildPipelineRunFinishTimeStr, 10, 64)
+		buildPipelineRunFinishTime := time.Unix(buildPipelineRunFinishTimeInt, 0)
+		buildPipelineRunFinishTimeMeta := &metav1.Time{Time: buildPipelineRunFinishTime}
+
+		go metrics.RegisterRunningSnapshot(*buildPipelineRunFinishTimeMeta, snapshotInProgressTime)
+		go metrics.RegisterIntegrationResponse(snapshot.GetCreationTimestamp(), snapshotInProgressTime)
```

Comment:
I think that this metric should be set even if there isn't a `BuildPipelineRunFinishTimeLabel` set on the snapshot, so it can be set outside this `if` condition.

Reply:
Updated. Thanks!

Reply:
I may be missing something, but the `RegisterIntegrationResponse` call is still showing under the `if helpers.HasLabel(snapshot, BuildPipelineRunFinishTimeLabel)` block for me, can you check this again @jinqi7 ?

Reply:
The file was missed to be committed.  How about now?

Reply:
Looks good now, thanks